### PR TITLE
Respond NOT_FOUND when partition cannot be found

### LINKED ISSUE #3961: SetVariables Command can cause GRPC 13: INTERNAL ERROR / GRPC 4: TIME OUT
**Describe the bug**
Issuing this command to the broker causes GRPC 13: Internal Error on 0.22, and GRPC 4 DEADLINE EXCEEDED on 0.23.0-alpha1:

(Node.js code)

```
await zbc.setVariables({
	elementInstanceKey: '0', 
	local: false,
	variables: {
		conditionVariable: false,
	},
})
```

**To Reproduce**
Send a SetVariables command for a non-existent elementInstance.

**Expected behavior**
I would expect 5 NOT FOUND. And does it have to wait to the gateway timeout...?

Is it waiting for that element to appear?

If it is, I'll need to document this and wrap the error to have it make sense for the user in the client.

**Environment:**
 - OS: Docker
 - Zeebe Version: Camunda Cloud, and 0.23.0-alpha1 in Docker.
 - Configuration: nada in Docker.

### PR DESCRIPTION
**Pull request by korthout**
## Description

When a request from a client refers to an elementInstanceKey that does
not encapsulate a partition (or encapsulates a non-existing partition),
then the gateway now responds with NOT_FOUND.

The gateway simply checks that the partitionId is part of the known partitions of
the topology.

## Related issues

closes #3961

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1313da6c44575b5c89b5f6299fe3f7d514306413
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d6b25dc9d70bdb22ca883cd0bea8bdcbe45fd80e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+package io.zeebe.gateway.cmd;
+
+/**
+ * Represents an exceptional error that occurs when a partition can not be found. For example this
+ * can happen when the element instance key does not refer to a known partition.
+ */
+public class PartitionNotFoundException extends ClientException {
+
+  public PartitionNotFoundException(final int partitionId) {
+    super(String.format("Unknown partition '%d'", partitionId));
```
Comment (deepthidevaki) [inline gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java]: I suggest to have a more detailed message. Something like `Expected to send request to partition '0', but partition '0' unknown in topology `. https://github.com/zeebe-io/zeebe/wiki/Error-Guidelines

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:+1: Looks good. One small comment.

**Review by jwulf — CHANGES_REQUESTED** [review]

File: gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwulf) [reply gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java]: If we can think of / or discover other cases where the user sees this error and this text is misleading, we can adjust.

**Review by deepthidevaki — APPROVED** [review]

File: gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClient.java [hunk 2]
```diff
@@ -25,6 +25,8 @@
       BrokerResponseConsumer<T> responseConsumer,
       Consumer<Throwable> throwableConsumer);
 
+  <T> ActorFuture<BrokerResponse<T>> sendRequest(BrokerRequest<T> request, Duration requestTimeout);
```
Comment (korthout) [inline gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClient.java]: I reverted the removal of this method after discussing it with @npepinpe.

The method now only exists for a single test, but it was easier just to revert the removal of the method than to rewrite the test. A more thorough refactoring of the BrokerClient is needed anyways.

File: gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (korthout) [reply gateway/src/main/java/io/zeebe/gateway/cmd/PartitionNotFoundException.java]: @jwulf I've applied your comment to the code. Can you take another look?

**Review by jwulf — APPROVED** [review]
Good work!

**Pull request merged**